mitocw/edx-platform
# Push Course info to CCX Connector

This PR introduces the `ccxcon` app a "hook" to an external service, the CCX Connector, used to interact with CCX Courses.

This new app has been implemented inside the `openedx.code.djangoapps` package because in the long term it needs to be accessible both by `lms` and `cms` apps.

A new course advanced setting has been added to specify the URL of the remote service.

How to test this PR:
1- add an instance of `openedx.core.djangoapps.ccxcon.models.CCXCon` with the URL of the external service, the oauth client id and client secret
2- in CMS, add the **same URL of point 1** to the `CCX Connector URL` field
3- check in the remote CCX Connector that a new entry has been created for the course

NOTE: the functionality of this PR is dependent on the implementation of mitodl/ccxcon#70

Fixes #128 
Fixes #118
